src/GTP.cpp

```diff
@@ -218,7 +219,9 @@ bool GTP::execute(GameState & game, std::string xinput) {
         gtp_printf(id, PROGRAM_NAME);
         return true;
     } else if (command == "version") {
-        gtp_printf(id, PROGRAM_VERSION);
+        size_t filename_start = cfg_weightsfile.find_last_of("/\\") + 1;
+        filename_start = filename_start % cfg_weightsfile.length();
+        gtp_printf(id, "%s with weightfile %s", PROGRAM_VERSION, cfg_weightsfile.substr(filename_start, 8).c_str());
```

Comment:
can you use basename(cfg_weightsfile)

Reply:
substr does not throw if you request a length longer than the remaining characters.  From cppreference.com: "Number of characters to include in the substring (if the string is shorter, as many characters as possible are used)."

Reply:
I apologize, I double checked some spec (wonder what I was looking at) because I had thought it did allow extra.